packages/core/src/components/menu/menuItem.tsx

```diff
@@ -77,6 +77,11 @@ export interface IMenuItemProps extends ActionProps, LinkProps {
      */
     labelElement?: React.ReactNode;
 
+    /**
+     * Props to spread to the `Menu` within the `Popover`.
+     */
+    menuProps?: Partial<MenuProps> & object;
```

Comment:
we don't need `& object`... that's a weird quirk from an older version of TypeScript that's no longer necessary.

Reply:
also I think we should call this prop `submenuProps`

```suggestion
    submenuProps?: Partial<MenuProps>;
```

Reply:
I agree with the name change.

So the issue with just `Partial<MenuProps>` is that `MenuProps` doesn't have the commonly used prop `data-testid`. Any Ideas on how we can get that in there?

Reply:
hmm, there's no good way to do that in the object type. `data-*` attributes work in JSX because they're built into React. I guess you can leave the `& object` in this type and leave a note in the JSDoc about why it's there.

Reply:
How would you describe that?
    /**
     * Props to spread to the child `Menu` component if this item has a submenu.
     * `& object` allows for `data-*` attributes to be passed.
     */